### PR TITLE
Add documentation for viadot 2.0 with new repository structure

### PR DESCRIPTION
**Pull request by djagoda881**
## Summary

Added docs which contains information about repository and guidelines how to use it

### CODE REVIEW

File: docs/user_guide/adding_prefect_flow.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+from prefect import flow
+
+@flow(
+    name="extract--postgresql--adls",
+    description="Extract data from PostgreSQL database and load it into Azure Data Lake.",
+    retries=1,
+    retry_delay_seconds=60,
+)
+def postgresql_to_adls()
+    # Logic of our flow.
```
Comment (trymzet) [inline docs/user_guide/adding_prefect_flow.md:93]: I would include the best practice example here, same as in the task example. So show how to call  `postgresql_to_df()` and `df_to_adls()` here.

File: docs/user_guide/adding_prefect_flow.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+```
+
+At the end, add tests for the specified flow in `viadot/tests/orchestration/prefect_tests/flows`. Your PR will be accepted only if the test coverage is greater than or equal to 80%.
```
Comment (trymzet) [inline docs/user_guide/adding_prefect_flow.md:96]:
```suggestion
At the end, add tests for the specified flow in `viadot/tests/integration/orchestration/flows/test_<your_flow_name>.py`.
```

**Review by trymzet — CHANGES_REQUESTED** [review]

File: docs/user_guide/adding_source.md [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
+When you are done with the source, remember to import it in the `__init__.py` file. 
+
+```python
+# inside source/__init__.py
+
+from .postgresql import PostgreSQL
+```
+
+And that's all you need to know to create your own sources in the viadot repository. You can also take a look at any existing source like [MiniIO](https://github.com/dyvenia/viadot/blob/2.0/src/viadot/sources/minio.py) as a reference.
```
[the remark this answers is not in the document]
Comment (djagoda881) [reply docs/user_guide/adding_source.md:56]: [📝 Added description about to_df() in adding_source.md](https://github.com/dyvenia/viadot/pull/929/commits/998882a3ef13dbb812da67330ca649f1eda51b88)

File: docs/user_guide/adding_prefect_flow.md (hunk 1 above)
Comment (djagoda881) [reply docs/user_guide/adding_prefect_flow.md]: [📝 Improved docs in adding_prefect_flow.md](https://github.com/dyvenia/viadot/pull/929/commits/6c2955f2cda853850d7e9cc82c8d492de3f1a65a)

File: docs/user_guide/adding_prefect_flow.md [hunk 4]
```diff
@@ -0,0 +1,96 @@
+
+Let's assume that we've finished adding our source `PostgresSQL`. We want to use it in our flow `postgressql_to_adls`. This flow will take a table from the PostgreSQL database using our previously defined source and upload it to the Azure Data Lake.
+We will have to create a task that will take our data from the PostgreSQL database and put it into flow, where the task sending data to the Azure Data Lake already exists.
```
[the remark this answers is not in the document]
Comment (djagoda881) [reply docs/user_guide/adding_prefect_flow.md]: [📝 Improved docs in adding_prefect_flow.md](https://github.com/dyvenia/viadot/pull/929/commits/6c2955f2cda853850d7e9cc82c8d492de3f1a65a)

File: docs/user_guide/adding_prefect_flow.md [hunk 5]
```diff
@@ -0,0 +1,96 @@
+
+Let's assume that we've finished adding our source `PostgresSQL`. We want to use it in our flow `postgressql_to_adls`. This flow will take a table from the PostgreSQL database using our previously defined source and upload it to the Azure Data Lake.
```
[the remark this answers is not in the document]
Comment (djagoda881) [reply docs/user_guide/adding_prefect_flow.md]: [📝 Improved docs in adding_prefect_flow.md](https://github.com/dyvenia/viadot/pull/929/commits/6c2955f2cda853850d7e9cc82c8d492de3f1a65a)

File: docs/index.md [hunk 6]
```diff
@@ -3,21 +3,11 @@
 [![build status](https://github.com/dyvenia/viadot/actions/workflows/build.yml/badge.svg)](https://github.com/dyvenia/viadot/actions/workflows/build.yml)
 [![formatting](https://img.shields.io/badge/code%20style-black-000000.svg)](https://github.com/psf/black)
 [![codecov](https://codecov.io/gh/Trymzet/dyvenia/branch/main/graph/badge.svg?token=k40ALkXbNq)](https://codecov.io/gh/Trymzet/dyvenia)
```
[the remark this answers is not in the document]
Comment (djagoda881) [reply docs/index.md]: [📝 Removed badges from index.md](https://github.com/dyvenia/viadot/pull/929/commits/9973d8f7ca52a694dc7e060b9c1a4af02ab44f77)

File: docs/user_guide/docker_containers.md [hunk 7]
```diff
@@ -0,0 +1,45 @@
+Using docker containers during development is not longer required. If you already installed [Rye](https://rye.astral.sh/) all of dependecies should be installed.
+If you still want to use containers you have to build it using `docker-compose`.
```
[the remark this answers is not in the document]
Comment (djagoda881) [reply docs/user_guide/docker_containers.md]: [📝 Added Advanced Usage section](https://github.com/dyvenia/viadot/pull/929/commits/b13a8b14b764d929a0925920bbd663f4ecc2172c)

File: docs/user_guide/rye.md [hunk 8]
```diff
@@ -0,0 +1,3 @@
+Rye is a comprehensive project and package management solution for Python. Rye provides a unified experience to install and manage Python installations, `pyproject.toml` based projects, dependencies and virtualenvs seamlessly. It's designed to accommodate complex projects and monorepos and to facilitate global tool installations.
+
+If you are working as a developer in viadot repository, this tool is required to be setup. All instructions about how to install and use rye tool can be found [here](https://rye.astral.sh/).
```
[the remark this answers is not in the document]
Comment (djagoda881) [reply docs/user_guide/rye.md]: [📝 Moved docs about Rye into CONTRIBUTING.md](https://github.com/dyvenia/viadot/pull/929/commits/27245785dc560c7d9575b2f5fafd7c068352ddb9)

File: docs/user_guide/docker_containers.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (djagoda881) [reply docs/user_guide/docker_containers.md]: [📝 Moved docker tutorial section form docs to CONTRIBUTING.md](https://github.com/dyvenia/viadot/pull/929/commits/3302e4343d110a263f29323e37d0e7bfc5259939)

**Review by trymzet — CHANGES_REQUESTED** [review]
Looks good, just 2 more small things

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -4,6 +4,13 @@
 
 Follow the instructions in the [README](./README.md) to set up your development environment.
 
+### Rye
+
+Rye is a comprehensive project and package management solution for Python. Rye provides a unified experience to install and manage Python installations, `pyproject.toml` based projects, dependencies and virtualenvs seamlessly. It's designed to accommodate complex projects and monorepos and to facilitate global tool installations.
+
+If you are working as a developer in viadot repository, this tool is required to be setup. All instructions about how to install and use rye tool can be found [here](https://rye.astral.sh/).
```
[the remark this answers is not in the document]
Comment (djagoda881) [reply CONTRIBUTING.md]: [📝 Removed Rye description from CONTRIBUTING.md](https://github.com/dyvenia/viadot/pull/929/commits/93c70804fc7b6673e3365202c834ce556b83ca7c)

**Review by trymzet — APPROVED** [review]

**Pull request merged**